paderbox/utils/random_utils.py

```diff
@@ -18,80 +20,325 @@ def str_to_random_state(string):
 
 
 def _force_correct_shape(f):
-    """ This decorator sets the seed and fix the snr.
+    """decorator allowing to pass shape as tuple as well
+
+    Args:
+        f: wrapped function
+
+    Returns:
 
-    :param f: Function to be wrapped
-    :return: noise_signal
     """
     @wraps(f)
     def wrapper(*shape, **kwargs):
-        if not shape:
-            shape = (1,)
-        elif isinstance(shape[0], (tuple, list)):
+        if len(shape) > 0 and isinstance(shape[0], (tuple, list)):
```

Comment:
Does this fail, when `shape is None`?

Reply:
shape cannot be None here, can it? or do you mean when shape=(None,) ?

Reply:
Ah, you are right, my fault. The code is correct.